lib/grandpa/message_handler.go

```diff
@@ -299,20 +299,43 @@ func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{}
 	return eqvVoters
 }
 
+func verifyIfDescendantOfHighestFinalisedBlock(blockState BlockState, hash common.Hash) error {
```

Comment:
I would like to suggest changing this method name to `IsDescendantOHighestFinalisedBlock` and instead of returning an `error` return `(bool, error)`, makes easier to understand what the function does IMO

Reply:
Yes but then isn't the boolean just going to be the same as `err == nil`?

Reply:
Not really, there is a case where the hash is not a descendant of the highest finalized block, so in this case, there is no error we can return nil but for me does not sounds correct to return nil when the value is a bool

Reply:
Existing way is better.
In case verification fails we want to return error.

> Not really, there is a case where the hash is not a descendant of the highest finalized block, so in this case, there is no error and we can return nil but for me does not sounds correct to return nil when the right value is a bool

returning bool will just add extra code for no reason. In case of verification failure, I would have to convert the boolean value again to an error. So, returning nil like mentioned above is not an option.